### PR TITLE
Introduce vela as a kubectl plugin

### PR DESCRIPTION
**Pull request by yangsoon**
fix https://github.com/oam-dev/kubevela/issues/1468

introduce vela as a kubectl plugin, so we can use features provided by vela through kubectl.

```bash
kubectl vela dry-run -f app.yaml
kubectl vela live-diff -f app.yaml -r first-vela-app-v1
kubectl vela show webservice
```

- [x] integrate dry-run, live-diff and vela show for the plugin
- [x] add test

### CODE REVIEW

**Review by chengshiwen — COMMENTED** [review]
I have developed kubectl plugin, so there are some things to pay attention to:

1. Consider whether to use [cli-runtime](https://github.com/kubernetes/cli-runtime/) which is a set of packages to share code with kubectl for printing output or [sharing command-line options](https://godoc.org/k8s.io/cli-runtime/pkg/genericclioptions). By using the [genericclioptions](https://godoc.org/k8s.io/cli-runtime/pkg/genericclioptions) package from `cli-runtime`, you can support the global command-line flags listed in `kubectl options` (e.g. `--kubeconfig`, `--context` and many others) in your plugins.

2. Import authentication plugins (Go)
By default, plugins that use `client-go` cannot authenticate to Kubernetes clusters on many cloud providers. To address this, include the following import in your plugin:
```
import _ "k8s.io/client-go/plugin/pkg/client/auth"
```

3. The cross-platform versions of `kubectl-vela` need to be submitted to [krew-index](https://krew.sigs.k8s.io/plugins/), but the current Makefile does not support

4. Another question is that why not just reuse the existing `vela` binary? Just integrate `dry-run` and `live-diff` features into one `vela`? `krew` will help you rename `vela` to `kubectl-vela`

**Comment by wonderflow** [comment]
@chengshiwen Cool! The suggestion is really professional! 

Let me answer the last question. 

The `vela` binary is a pure developer facing tool which based on Appfile without any knowledge of K8s. This is more like a reference implementation for platform builder users of KubeVela.

While in K8s community, who can easily explain the developer user story to those DevOps who can access K8s easily but don't want to understand it deeply. So they can experience  KubeVela by directly using Application Object(yaml) and help to promote KubeVela to their developers. So we build `kubectl-vela`.

**Comment by yangsoon** [comment]
@chengshiwen  Thank you very much for your valuable experience, I will improve this part of the work.

File: pkg/plugin/cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		NewLiveDiffCommand(commandArgs, ioStream),
+		NewCapabilityShowCommand(commandArgs, ioStream),
+		NewVersionCommand(),
+		NewHelpCommand(),
+	)
+
+	// this is for mute klog
+	fset := flag.NewFlagSet("logs", flag.ContinueOnError)
+	klog.InitFlags(fset)
+	_ = fset.Set("v", "-1")
```
Comment (wonderflow) [inline pkg/plugin/cli/cli.go:72]: why?

**Review by wonderflow — APPROVED** [review]
LGTM

**Review by wonderflow — APPROVED** [review]
LGTM after CI

**Pull request merged**